activerecord/lib/active_record/connection_adapters/connection_specification.rb

```diff
@@ -179,7 +179,14 @@ def spec(config)
           end
 
           adapter_method = "#{spec[:adapter]}_connection"
-          ConnectionSpecification.new(spec, adapter_method)
+
+          id ||=
+            if config.is_a?(Symbol)
+              config.to_s
+            else
+              "primary"
```

Comment:
Something really irks me about these two conditionals. Partly that we can't know if the inner symbol check is run at all, or just the special treatment of symbols altogether. Haven't got anything more concrete right now, will let it simmer a bit 😁